src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp

```diff
@@ -387,6 +394,23 @@ ov::npuw::CompiledModel::CompiledModel(const std::shared_ptr<ov::Model>& model,
     reset_io();
 }
 
+void ov::npuw::CompiledModel::fill_weights_bank(const std::size_t idx) {
+    LOG_VERB("Filling weights bank for Subgraph[" << idx << "]...");
+    LOG_BLOCK();
+    auto& comp_model_desc = m_compiled_submodels[idx];
```

Comment:
I think you can explicitly put asserts here that
1. A submodel is not optimized out
2. A submodel is a function

Reply:
Done

Reply:
Well, not entirely